### ISSUE TITLE
Enhancement: put Appplication Form in its own class

### ISSUE THREAD

**Issue by wernerkrauss** [dynamic/silverstripe-jobs#30]
In my current project I heavily adjusted ApplicationForm in JobController, now I need the same form with a dropdown of available jobs in the JobCollectionController.

It'd be cleaner, if ApplicationForm was encapsulated in a subclass of `Form`, then all the global modifications could be done in one place and I could adjust it slightly per controller.

**Comment by muskie9** [comment]
@wernerkrauss this is a very timely ticket as we have been discussing this and other similar adjustments to make the module more extensible. I like the idea of the form being a subclass and encapsulated for the very reason you mentioned.

We have a project coming up which we were planning on putting quite a bit of work into the module but if you're open to taking a shot at making this type of change to speed things up we're happy to review so you're not waiting on this feature.

**Comment by muskie9** [comment]
@wernerkrauss I've started #31 as a super rough PR. There are a few things that may be out of scope for this that I noticed as broken and haven't been able to test a submission quite yet.

This is also in a Silverstripe 4.13 installation. Some of these changes could require a new major version of the module, but if nothing is breaking between 4 and 5 we could possibly support both in the new major version. Open to feedback/suggestions.
